src/services/shop/shop-router.ts

```diff
@@ -42,148 +43,423 @@ shopRouter.get(
-        const itemExists = (await Models.ShopItem.findOne({ name: details.name })) ?? false;
-        if (itemExists) {
-            return res.status(StatusCode.ClientErrorConflict).send(ShopItemAlreadyExistsError);
-        }
+            // Retrieve the user's profile
+            const profile = await Models.AttendeeProfile.findOne({ userId: order.userId });
+            if (!profile) {
+                return res.status(StatusCode.ServerErrorInternal).send(ShopInternalError);
+            }
+
+            let totalPointsRequired = 0;
+
+            // Loop through items and check availability and price
+            for (let i = 0; i < order.items.length; i++) {
+                const item = await Models.ShopItem.findOne({ itemId: order.items[i] });
+                if (!item) {
+                    return res.status(StatusCode.ClientErrorNotFound).send(ShopItemNotFoundError);
+                }
+
+                const quantity = order.quantity?.[i] ?? 0; // Default to 0 if undefined
+                totalPointsRequired += quantity * item.price;
 
-        const newItem = await Models.ShopItem.create(shopItem);
-        const withoutInstances = {
-            ...newItem.toObject(),
-            instances: undefined,
-        };
+                // Check if requested quantity is available
+                if (quantity > item.quantity) {
+                    return res.status(StatusCode.ClientErrorBadRequest).send(ShopInsufficientQuantityError);
+                }
+            }
 
-        return res.status(StatusCode.SuccessOK).send(withoutInstances);
+            // Check if the user has enough points for the order
+            if (!profile || profile.points < totalPointsRequired) {
+                return res.status(StatusCode.ClientErrorPaymentRequired).send(ShopInsufficientFundsError);
+            }
+
+            // Update the inventory and user points
+            for (let i = 0; i < order.items.length; i++) {
+                const item = await Models.ShopItem.findOne({ itemId: order.items[i] });
+                const quantity = order.quantity?.[i] ?? 0;
+                if (!item) {
+                    return res.status(StatusCode.ClientErrorNotFound).send(ShopItemNotFoundError);
+                }
+
+                // Deduct item quantity from stock
+                await Models.ShopItem.updateOne({ itemId: order.items[i] }, { $inc: { quantity: -quantity } });
+
+                // Deduct points from user profile
+                await updatePoints(order.userId, -(quantity * item.price));
+            }
+
+            // Clear the user's order from the cart
+            await Models.ShopOrder.deleteOne({ userId });
+            return res.status(StatusCode.SuccessOK).json(order);
+        } catch (error) {
+            console.error("Error processing order:", error);
+            return res.status(StatusCode.ServerErrorInternal).json(ShopInternalError);
+        }
```

Comment:
Don't wrap everything in a try catch, the error middleware will handle any undefined errors as internal errors with proper logging.